dask_sql/physical/rex/core/call.py

```diff
@@ -656,7 +657,10 @@ def timestampadd(self, unit, interval, df: SeriesOrScalar):
         interval = int(interval)
         if interval < 0:
             raise RuntimeError(f"Negative time interval {interval} is not supported.")
-        df = df.astype("datetime64[ns]")
+        if is_timestamp_nano(df):
+            df = df.astype("datetime64[s]")
+        else:
+            df = df.astype("datetime64[ns]")
```

Comment:
Could condense this to something like

```suggestion
        df = df.astype("datetime64[s]" if is_timestamp_nano(df) else "datetime64[ns]")
```